internal/gitprovider/azure/azure.go

```diff
@@ -181,6 +181,83 @@ func (p *provider) ListPullRequests(
 	return pts, nil
 }
 
+// MergePullRequest implements gitprovider.Interface.
+func (p *provider) MergePullRequest(
+	ctx context.Context,
+	id int64,
+	opts *gitprovider.MergePullRequestOpts,
+) (*gitprovider.PullRequest, error) {
+	gitClient, err := adogit.NewClient(ctx, p.connection)
+	if err != nil {
+		return nil, fmt.Errorf("error creating Azure DevOps client: %w", err)
+	}
+
+	if opts == nil {
+		opts = &gitprovider.MergePullRequestOpts{}
+	}
+
+	// Set merge strategy
+	mergeStrategy := adogit.GitPullRequestMergeStrategyValues.NoFastForward // Default merge commit
+	if opts.MergeMethod != "" {
+		switch strings.ToLower(opts.MergeMethod) {
+		case "squash":
+			mergeStrategy = adogit.GitPullRequestMergeStrategyValues.Squash
+		case "rebase":
+			mergeStrategy = adogit.GitPullRequestMergeStrategyValues.Rebase
+		default:
+			// Default to NoFastForward for "merge" or any other method
+			mergeStrategy = adogit.GitPullRequestMergeStrategyValues.NoFastForward
+		}
+	}
```

Comment:
It seems all the providers have code similar to this. I'm wondering why we're bothering with this given that we're not surfacing this option in the step's configuration.

If our first iteration of the merge step is "merge or die," shouldn't we just merge without specifying and accept the remote repo's default merge strategy?